src/classes/UTIL_OrgTelemetry_SVC.cls

```diff
@@ -33,21 +33,218 @@
 * @group Utilities
 * @description Return some basic org and NPSP telemetry data to the License Management Application to assist with
 * better understanding of how our customers use our application.
+* This logic is called in two places:
+*   1) When Customizable Rollups is enabled or disabled
+*   2) During the Post-Install Script as part of a Package Update
+* Note: FeatureManagement operations are subject to MIXED_DML errors and must be conducted asynchronously
+* Uses "without sharing" so that it has full rights to query all records in all objects. This is needed to ensure a
+* consistent return of telemetry data even if the process is called by different users with different permissions.
+* Specific PII info is never queried or returned by this class.
 */
 public without sharing class UTIL_OrgTelemetry_SVC {
 
     /**
-    * @description Submit specific feature enablement statistics back to the License Management Org to assist
-    * with understanding how our customers are using our functionality. There's no way to unit test this though.
-    * Each feature below must be matched to a SubscriberToLmo feature of the same name.
-    * This should called in two places: 1) When the Settings page is loaded; 2) When a tracked area is saved.
-    * FeatureManagement operations are subject to MIXED_DML errors and must be conducted asynchronously
-    */
+     * @description A complete list of the Telemetry Types (at a high level) that are handled by this service class.
+     * These are used by the batch job to process feature parameters in asynchronous groups to avoid governor
+     * limit issues. The concept is that by separating each of these into smaller groupings that are processed
+     * as separate child batch iterations, the entire process can complete even if for some reason one of the queries
+     * against a table (such as a really huge Opportunity object) were to fail.
+     */
+    public Enum TelemetryTypeEnum {
```

Comment:
No need for the Enum in the variable name.

Reply:
Agreed